xwrf/funcs_from_wrfpython.py

```diff
@@ -0,0 +1,52 @@
+def _destag_single_var(datavar, stagger_dim=None, unstag_dim_name=None):
+
+    # "de-stagger" function for a single xarray wrf data variable
+    # returns the de-staggered and variable with renamed dimension
+
+    # based off of the wrf-python destagger function
+    # https://github.com/NCAR/wrf-python/blob/22fb45c54f5193b849fdff0279445532c1a6c89f/src/wrf/destag.py
+
+    # datavar: xarary data array
+    # stagger_dim: <str, optional> dimension to unstagger. fx will guess based on name (ends in "_stag")
+    # unstag_dim_name: <str, optional> argument used to rename the dimension.
+    #                  example would be "west_east" for "west_east_stag"
+    #                  by default the dimenions will be renamed the text in front
+    #                  of "_stag" from the "stagger_dim" field
+
+    # ------------------------------------------------------
+
+    # get the coordinate to unstagger
+    # option 1) user has provided the dimension
+    if stagger_dim:
+        # check that the user-passed in stag dim is actually in there
+        assert stagger_dim in datavar.dims, AttributeError(f'{stagger_dim} not in {datavar.dims}')
```

Comment:
We could combine these lines:

```suggestion
    # check that the user-passed in stag dim is actually in there
    if stagger_dim and stagger_dim not in datavar.dims:
        raise valueError(f'{stagger_dim} not in {datavar.dims}')
```